lib/puppet_x/puppetlabs/powershell_manager.rb

```diff
@@ -0,0 +1,123 @@
+require 'open3'
+require 'thread'
+
+module PuppetX::Dsc
+  class PowerShellManager
+    attr_accessor :return_code
+
+    @@instances = {}
+
+    def self.instance(cmd)
+      @@instances[:cmd] ||= PowerShellManager.new(cmd)
+    end
+
+    def initialize(cmd)
+      process_init_lock = Mutex.new
+      process_started = ConditionVariable.new
+
+      @stdout_lock = Mutex.new
+      @stdout_written = ConditionVariable.new
+
+      @stdout_capture = []
+      stdout_monitor_thread = nil
+
+      process_init_lock.synchronize do
+        @runner_thread = Thread.new do
+          Open3::popen2(cmd) do |stdin, stdout, proc_thread|
+            Puppet.debug "#{Time.now} #{cmd} is running as pid: #{proc_thread.pid}"
+
+            @stdin = stdin
+            @stdout = stdout
+            stdin.sync = true
+            stdout.sync = true
```

Comment:
During rspec runs, the process has started and already exited by the time it hits this line of code, so it hangs once it gets down to waiting on the process to finish opening.